packages/flutter/lib/src/services/image_stream.dart

```diff
@@ -274,3 +288,163 @@ class OneFrameImageStreamCompleter extends ImageStreamCompleter {
+/// scheduled).
+///
+/// See the following timeline example:
+///
+/// | Time | Event                                      | Comment                   |
+/// |------|--------------------------------------------|---------------------------|
+/// | t1   | App frame scheduled (image frame A posted) |                           |
+/// | t2   | App frame scheduled                        |                           |
+/// | t3   | App frame scheduled                        |                           |
+/// | t4   | Image frame B decoded                      |                           |
+/// | t5   | App frame scheduled                        | t5 - t1 < frameB_duration | 
+/// | t6   | App frame scheduled (image frame B posted) | t6 - t1 > frameB_duration |
+///
+class MultiFrameImageStreamCompleter extends ImageStreamCompleter {
+  /// Creates a image stream completer.
+  ///
+  /// Immediately starts decoding the first image frame when the codec is ready.
+  ///
+  /// [codec] is a future for an initialized [ui.Codec] that will be used to
+  /// decode the image.
+  /// [scale] is the linear scale factor for drawing this frames of this image
+  /// at their intended size.
+  MultiFrameImageStreamCompleter({
+    @required Future<ui.Codec> codec,
+    @required double scale,
+    InformationCollector informationCollector
+  }) :
+    _informationCollector = informationCollector,
+    _scale = scale,
+    _framesEmitted = 0,
+    _timer = null
+  {
+    assert(codec != null);
+    codec.then<Null>(_onCodecReady, onError: (dynamic error, StackTrace stack) {
+      FlutterError.reportError(new FlutterErrorDetails(
+        exception: error,
+        stack: stack,
+        library: 'services',
+        context: 'resolving an image codec',
+        informationCollector: informationCollector,
+        silent: true,
+      ));
+    });
+  }
+
+  ui.Codec _codec;
+  final double _scale;
+  final InformationCollector _informationCollector;
+  ui.FrameInfo _nextFrame;
+  // When the current was first shown.
+  Duration _shownTimestamp;
+  // The requested duration for the current frame;
+  Duration _frameDuration;
+  // How many frames have been emitted so far.
+  int _framesEmitted;
+  Timer _timer;
+
+  void _onCodecReady(ui.Codec codec){
+    print('codec ready');
+    _codec = codec;
+    _decodeNextFrameAndSchedule();
+  }
+
+  void _onAppFrame(Duration timestamp) {
+    if (!_hasActiveListeners)
+      return;
+    if (_isFirstFrame() || _hasFrameDurationPassed(timestamp)) {
+      _emitFrame(new ImageInfo(image: _nextFrame.image, scale: _scale));
+      _shownTimestamp = timestamp;
+      _frameDuration = _nextFrame.duration;
+      _nextFrame = null;
+      int completedCycles = (_framesEmitted / _codec.frameCount).floor();
+      if (_codec.repetitionCount == -1 || completedCycles <= _codec.repetitionCount) {
+        _decodeNextFrameAndSchedule();
+      }
+      return;
+    }
+    _timer = new Timer(_frameDuration - (timestamp - _shownTimestamp),
```

Comment:
this should take into account the timeDilation

Reply:
Looks like the timestamps handed to the app frame callbacks are already dilated... so we don't need to do anything here...

Reply:
Ohh you're right, this timer is not affected by dilation, multiplied by time dilation.